src/components/offer-card/OfferCard.jsx

```diff
@@ -1,3 +1,171 @@
-const OfferCard = () => {}
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import Avatar from '@mui/material/Avatar'
+import Rating from '@mui/material/Rating'
+import Chip from '@mui/material/Chip'
+import IconButton from '@mui/material/IconButton'
+import BookmarkBorderOutlinedIcon from '@mui/icons-material/BookmarkBorderOutlined'
+import Button from '@mui/material/Button'
+import LanguageIcon from '@mui/icons-material/Language'
+import StarIcon from '@mui/icons-material/Star'
+import { useTranslation } from 'react-i18next'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import { Link } from 'react-router-dom'
+import { styles } from '~/components/offer-card/OfferCard.styles.js'
+
+const OfferCard = ({ offer, addToList, sendMessagePopup }) => {
+  const { t } = useTranslation()
+  const { isMobile, isTablet } = useBreakpoints()
+  const getShortAuthorName = () => {
+    const shortLastName = offer.author.lastName.split('')[0]
+    return `${offer.author.firstName} ${shortLastName}.`
+  }
+  const getAuthorName = () => {
+    return `${offer.author.firstName} ${offer.author.lastName}`
+  }
+  const getReviews = () => {
+    const reviews =
+      offer?.author?.totalReviews?.student + offer?.author?.totalReviews?.tutor
+    return t('tutorProfilePage.reviews.reviewsCount', { count: reviews })
+  }
+  const getAuthorlanguages = () => {
+    return offer.languages.join(', ')
+  }
+  const getPrice = () => {
+    return `${offer.price} ${t('common.uah')}`
+  }
+  const languages = (
+    <Box sx={styles.languagesBox}>
+      <LanguageIcon sx={styles.languageIcon}></LanguageIcon>
+      <Typography>{getAuthorlanguages()}</Typography>
+    </Box>
+  )
+  const getTruncatedSubject = () => {
+    return offer.subject.name.length > 15
+      ? offer.subject.name.substring(0, 15) + '...'
+      : offer.subject.name
+  }
```

Comment:
It can be like the common pure function. Maybe we can use it in different components.
```suggestion
  const getTruncatedSubject = (text, maxSize) => {
    return text > maxSize
      ? text .subject.name.substring(0, maxSize) + '...'
      : text 
  }
```